sw/device/silicon_creator/rom_ext/imm_section/utils.bzl

```diff
@@ -64,3 +74,83 @@ def create_imm_section_targets(name, src):
         data = [object_target_name],
         alwayslink = 1,
     )
+
+_RELEASE_BUILD_HEADER = """\
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load(
+    "//sw/device/silicon_creator/rom_ext/imm_section:utils.bzl",
+    "create_imm_section_targets",
+)
+
+package(default_visibility = ["//visibility:public"])
+
+# Release version: {version}
+"""
+
+_RELEASE_BUILD_TARGET = """
+create_imm_section_targets(
+    name = "{name}",
+    src = "{filename}",
+)
+"""
+
+def _prepare_release_files(ctx):
+    build_contents = [
+        _RELEASE_BUILD_HEADER.format(version = IMM_SECTION_VERSION),
+    ]
+    build_file = ctx.actions.declare_file("BUILD")
+
+    files = [build_file]
+    for name, target in zip(ctx.attr.variants_keys, ctx.attr.variants_values):
+        bin = get_one_binary_file(target, field = "binary", providers = [SiliconBinaryInfo])
```

Comment:
perhaps drop a comment here saying why we only release the silicon exec_env binary? (i.e. b/c they are all the same and a test enforces that, perhaps call out the test too)